thunder/core/transforms.py

```diff
@@ -3743,8 +3743,8 @@ def backward_fn(saved_for_backward, cotangents):
             proxified if isinstance(entry, Proxy) else entry
             for proxified, entry in zip(flat_saves_proxified, flat_saves)
         ]
-        saved_for_backward = tree_unflatten(flat_filtered, saves_spec)
-        env = reconstruct_forward_env_for_backward(trace, saved_for_backward)
+        unproxiefied_saved_for_backward = tree_unflatten(flat_filtered, saves_spec)
```

Comment:
@IvanYashchuk embarrassingly I don't quite get how this would work with a `tree_map`... how would I be able to zip two pytree while still keep them as a pytree?